app/uk/gov/hmrc/cataloguefrontend/platforminitiatives/PlatformInitiativesController.scala

```diff
@@ -19,51 +19,86 @@ package uk.gov.hmrc.cataloguefrontend.platforminitiatives
 import play.api.data.Form
 import play.api.data.Forms.{mapping, optional, text}
 import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import uk.gov.hmrc.cataloguefrontend.connector.TeamsAndRepositoriesConnector
 import uk.gov.hmrc.cataloguefrontend.platforminitiatives.html.PlatformInitiativesListPage
 import uk.gov.hmrc.http.HeaderCarrier
 import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
 
 import javax.inject.{Inject, Singleton}
-import scala.concurrent.ExecutionContext
+import scala.concurrent.{ExecutionContext, Future}
 
 @Singleton
-class PlatformInitiativesController @Inject() (
-  mcc                           : MessagesControllerComponents,
-  platformInitiativesConnector  : PlatformInitiativesConnector,
-  platformInitiativesListPage   : PlatformInitiativesListPage
+class PlatformInitiativesController @Inject()
+( mcc                           : MessagesControllerComponents
+, platformInitiativesConnector  : PlatformInitiativesConnector
+, platformInitiativesListPage   : PlatformInitiativesListPage
+, teamsAndRepositoriesConnector : TeamsAndRepositoriesConnector
 )(implicit val ec: ExecutionContext)
   extends FrontendController(mcc) {
   implicit val hc: HeaderCarrier = HeaderCarrier()
 
-  def platformInitiatives(display: DisplayType): Action[AnyContent] =
-    Action.async { implicit request =>
-      platformInitiativesConnector.allInitiatives.map { initiative =>
-        PlatformInitiativesFilter.form
-          .bindFromRequest()
-          .fold(
-            formWithErrors => Ok(platformInitiativesListPage(
-              initiatives       = Seq(),
-              display           = display,
-              formWithErrors
-            )),
-            _ =>
-              Ok(platformInitiativesListPage
-              ( initiatives     = initiative,
-                display         = display,
+  def platformInitiatives(team: String = "", display: DisplayType): Action[AnyContent] = Action.async { implicit request =>
+    PlatformInitiativesFilter.form
+      .bindFromRequest()
+      .fold(
+        formWithErrors => Future.successful(Ok(platformInitiativesListPage
+        (initiatives  = Seq()
+        , display     = display
+        ,  team       = team
+        ,  teams      = Seq()
+        ,  formWithErrors
+        ))),
+        _ =>
+        request.uri match {
+            case uri if uri.contains("/platform-initiatives/") => {
```

Comment:
The uri parsing is brittle. It might be avoided if we just have a single route with optional team.
We should be able to just pull the optional team out of the form

Reply:
Amended following setting team to `Option[String]`